tests/unit/test_data_cleaning.py

```
@@ -0,0 +1,195 @@
+        np.testing.assert_array_almost_equal(result.tolist(), expected)
+
+
+class TestOutlierDetection:
+    """Tests for outlier detection and handling."""
+
+    def test_detect_outliers_iqr(self):
+        """Test detect_outliers using IQR method."""
+        test_series = pd.Series([1, 2, 3, 4, 5, 100])
+        mask, stats = detect_outliers(test_series, method='iqr')
+        
+        assert mask.iloc[5] == True
```

Ruff (0.14.13)

Avoid equality comparisons to `True`; use `mask.iloc[5]:` for truth checks

Replace with `mask.iloc[5]`

(E712)